src/components/Main/Map.jsx

```diff
@@ -0,0 +1,56 @@
+import React, { useRef, useEffect, useState } from "react";
+import mapboxgl from "!mapbox-gl"; // eslint-disable-line import/no-webpack-loader-syntax
```

Comment:
What this `!` syntax? I've never seen it before!

Reply:
I just copy paste this from [mapbox](https://docs.mapbox.com/help/tutorials/use-mapbox-gl-js-with-react/#add-mapbox-gl-js)

basically, this is the so-called "no webpack loader syntax". I am not pretty sure why they put this here (I guess this is for performance??), but you can find some detailed explanations [here](https://webpack.js.org/concepts/loaders/#configuration) and explain to me XD